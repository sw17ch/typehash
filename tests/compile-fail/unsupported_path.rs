//! Test that we can compile an empty struct.

use typehash::TypeString;

pub mod m {
    use super::*;
    #[derive(TypeString)]
    pub struct Other;
}

#[derive(TypeString)]
struct Empty {
    _other: m::Other,
}

fn main() {}