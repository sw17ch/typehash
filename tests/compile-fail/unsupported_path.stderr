error: The type referred to by field '_other' is not supported: type is not an ident: m :: Other.
  --> $DIR/unsupported_path.rs:11:10
   |
11 | #[derive(TypeString)]
   |          ^^^^^^^^^^
   |
   = note: this error originates in the derive macro `TypeString` (in Nightly builds, run with -Z macro-backtrace for more info)
